app/config/events.php

```diff
@@ -237,6 +251,51 @@
             '$description' => 'This event triggers when a function is updated.',
         ]
     ],
+    'messages' => [
+        '$model' => Response::MODEL_MESSAGE,
+        '$resource' => true,
+        '$description' => 'This event triggers on any messaging event.',
+        'create' => [
+            '$description' => 'This event triggers when a message is created.',
+        ],
+        'providers' => [
+            '$model' => Response::MODEL_PROVIDER,
+            '$resource' => true,
+            '$description' => 'This event triggers on any provider event.',
+            'create' => [
+                '$description' => 'This event triggers when a provider is created.',
+            ],
+            'update' => [
+                '$description' => 'This event triggers when a provider is updated.',
+            ],
+            'delete' => [
+                '$description' => 'This event triggers when a provider is deleted.'
+            ],
+        ],
```

Comment:
Let's move providers to the top level instead of nesting (contrary to the RFC), they're generic enough that we could potentially use them for other provider types like auth in future